### PR TITLE
enable direct routing to the send page

### LINKED ISSUE #15188: Can't use browser back button on the send screen
### Describe the bug

-

### Steps to reproduce

1. Go to "Send"
2. Fill out the send screen form and click "Next"
3. Once on the confirm screen, click the browser "Back" button
4. You will see an error message

https://user-images.githubusercontent.com/7499938/178381111-52556892-f98d-493f-85aa-2ca9969cb1e3.mp4

### Error messages or log output

_No response_

### Version

10.18.0

### Build type

_No response_

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by brad-decker**
## Explanation
When we refactored to not use global state for transaction data in the send flow, we made it a requirement to start a new draft transaction PRIOR to entering the send flow. This change allows us to gracefully handle situations that are edge cases such as when routing to the send screen directly, or if the user is in full screen mode and clicks back after moving to the confirm screen. Also added a test to cover this scenario.

## More Information
Fixes #15188 #15183 

## Manual Testing Steps

1. Start a send in full screen mode
2. Fill in amount and click next to move to confirmation screen
3. Click  browser back button
4. Send screen does not error. 
5. Pick a different recipient
6. Add an amount
7. Click next
8. See 2 confirmations, each of which is a fully formed transaction reflecting user's intent

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [x] Manual testing complete & passed
- [x] "Extension QA Board" label has been applied

### CODE REVIEW

File: ui/pages/send/send-header/send-header.component.js [hunk 1]
```diff
@@ -19,15 +20,19 @@ export default function SendHeader() {
...
   };
 
-  let title = asset.type === ASSET_TYPES.NATIVE ? t('send') : t('sendTokens');
+  let title = asset?.type === ASSET_TYPES.NATIVE ? t('send') : t('sendTokens');
 
-  if (stage === SEND_STAGES.ADD_RECIPIENT || stage === SEND_STAGES.INACTIVE) {
+  if (
+    draftTransactionExists === false ||
+    stage === SEND_STAGES.ADD_RECIPIENT ||
+    stage === SEND_STAGES.INACTIVE
```
Comment (darkwing) [inline ui/pages/send/send-header/send-header.component.js:34]: Since you used `.includes` on line https://github.com/MetaMask/metamask-extension/pull/15259/files#diff-5fd11e9ad27369ac2d1e53fc158bc9fd26efaf0b9224509843ce70d141bd13bfR97, do you want to use it here?

File: ui/pages/send/send.test.js [hunk 2]
```diff
@@ -163,6 +177,25 @@ describe('Send Page', () => {
...
+          ...baseStore.send,
+          currentTransactionUUID: null,
+        },
+      };
+      const store = configureMockStore(middleware)(modifiedStore);
+      const { getByPlaceholderText } = renderWithProvider(<Send />, store);
+      // Ensure that the send flow renders on the add recipient screen when
+      // there is no draft transaction.
+      expect(
+        getByPlaceholderText('Search, public address (0x), or ENS'),
```
Comment (darkwing) [inline ui/pages/send/send.test.js:194]: I know we use this in another place but selecting elements by placeholder always feels kinda weird.

Comment (darkwing) [reply ui/pages/send/send.test.js:194]: Probably just because I think of selecting elements by CSS paradigms, and you usually wouldn't select an element like this.  It's fine, just breaks my brain a little.

File: ui/pages/send/send-header/send-header.component.js (hunk 1 above)
Comment (brad-decker) [reply ui/pages/send/send-header/send-header.component.js]: Yup, good call out. Done in latest

**Review by adonesky1 — APPROVED** [review]
LGTM! Code looks good, and QA'd testing steps and issues linked

**Review by darkwing — APPROVED** [review]
